src/diffusers_interpret/generated_images.py

```diff
@@ -144,3 +144,18 @@ def show(self, width: Union[str, int] = "100%", height: Union[str, int] = "400px
         self.image_slider_iframe.width = width
         self.image_slider_iframe.height = height
         display.update(self.image_slider_iframe)
+
+    def gif(self, file_name = "diffusion_process.gif", duration = 400, show = True):
+        '''
+        Generate and display a GIF from the denoising process
+        '''
+        self[0].save(file_name,
+                     save_all = True,
+                     append_images = self,
+                     optimize = False,
+                     duration = duration,
+                     loop = 0)
+        if show == True:
+            return IPython.display.Image(file_name)
+        else:
+            return None

```

Comment:
since we imported as `from IPython import display as d`
Also, returning the display is not required 
```suggestion
        if show:
            d.Image(file_name)

```